library/core/src/fmt/num.rs

```diff
@@ -331,9 +331,15 @@ macro_rules! impl_Exp {
                     let rem = n % 10;
                     n /= 10;
                     exponent += 1;
-                    // round up last digit
-                    if rem >= 5 {
+                    // round up last digit, round to even on a tie
+                    if rem > 5 || (rem == 5 && n % 2 != 0) {
```

Comment:
Sorry I keep noticing things, but I think we're almost there...

The check for even here is missing whether we removed any additional non-zero digits earlier. e.g., for `{:.1e}` I expect:

* 124999 => 1.2e5 (rounded down)
* 125000 => 1.2e5 (tie rounded down to even)
* 125001 => 1.3e5 (rounded up)
  * This one gets treated like a tie right now and prints 1.2e5
* 134999 1.3e5 (rounded down)
* 135000 1.4e5 (tie rounded up to even)
* 135001 1.4e5 (rounded up)

I think `subtracted_precision > 1` would account for the extra non-zero digits? Please do add these values as tests.